x-pack/plugins/uptime/public/components/fleet_package/custom_fields.tsx

```diff
@@ -0,0 +1,414 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useContext, useEffect, useState, memo } from 'react';
+import { FormattedMessage } from '@kbn/i18n/react';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiForm,
+  EuiFormRow,
+  EuiFieldText,
+  EuiFieldNumber,
+  EuiSelect,
+  EuiSpacer,
+  EuiDescribedFormGroup,
+  EuiCheckbox,
+} from '@elastic/eui';
+import { ConfigKeys, DataStream, ISimpleFields, Validation } from './types';
+import { SimpleFieldsContext } from './contexts';
+import { TLSFields, TLSRole } from './tls_fields';
+import { ComboBox } from './combo_box';
+import { OptionalLabel } from './optional_label';
+import { HTTPAdvancedFields } from './http_advanced_fields';
+import { TCPAdvancedFields } from './tcp_advanced_fields';
+import { ScheduleField } from './schedule_field';
+
+interface Props {
+  typeEditable?: boolean;
+  isTLSEnabled?: boolean;
+  validate: Validation;
+}
+
+export const CustomFields = memo<Props>(
```

Comment:
This is a huge component, would have made sense to separate it into multiple for better maintenance and readability